src/viztracer/modules/snaptrace.c

```diff
@@ -898,22 +904,29 @@ snaptrace_dump(TracerObject* self, PyObject* args)
     // == Load the metadata first ==
     //    Process Name
     {
-        PyObject* current_process_method = PyObject_GetAttrString(multiprocessing_module, "current_process");
-        if (!current_process_method) {
-            perror("Failed to access multiprocessing.current_process()");
-            exit(-1);
-        }
-        PyObject* current_process = PyObject_CallObject(current_process_method, NULL);
-        if (!current_process_method) {
-            perror("Failed to access multiprocessing.current_process()");
-            exit(-1);
+        PyObject* process_name = NULL;
+        if (self->process_name) {
+            process_name = self->process_name;
```

Comment:
You are losing reference here. You need to increment the reference because you are decreasing it later.